static/js/narrow.js

```diff
@@ -268,23 +268,72 @@ export function activate(raw_operators, opts) {
         const adjusted_operators = [];
         let operators_changed = false;
 
+        function message_history_match_narrow(narrow_topic, narrow_stream_id, message) {
```

Comment:
I'd add a block comment at the top describing what the function does, and also move consider moving it to `message_edit.js`. Generally it's not a great pattern to define functions inside other functions that don't need to be there (E.g. because they access variables from the parent), and structurally this is a pure function of its 3 inputs.

(Which BTW we should probably split to have a separate `message_edit_ui.js` for the UI parts; for now I'd just place this near other pure functions that would not be moved in such a split)

Reply:
I moved that function to `message_edit.js` as you suggested with the comment block, though I feel like I also improved it's name to be clearer as well.